hagrid/products/models.py

```diff
@@ -71,13 +95,101 @@ class Variation(models.Model):
+    count_prio_bumped = models.BooleanField(default=False)
     availability = models.CharField(default=STATE_MANY_AVAILABLE, max_length=20, choices=AVAILABILITY_STATES)
+    crate_size = models.IntegerField(help_text="How many items of this variation come in a crate (overrides product setting)?", blank=True, null=True)
 
     def __str__(self):
         return "{} ({})".format(str(self.product), str(self.size))
 
+    @property
+    def availability_progress(self):
+        progress = 0
+        if self.count is not None and self.initial_amount:
+            progress = self.count / self.initial_amount * 100
+        return f"{progress:.1f}%"
+
+    @property
+    def is_count_reserved(self):
+        return self.count_reserved_until and self.count_reserved_until > timezone.now()
+
+    @property
+    def is_count_disabled(self):
+        return self.count_disabled_until and self.count_disabled_until > timezone.now()
+
+    @property
+    def computed_availability(self):
+        if self.count is not None and self.initial_amount:
+            if self.count == 0:
+                return Variation.STATE_SOLD_OUT
+
+            if self.count <= 2 or self.count < self.initial_amount * 0.1:
+                return Variation.STATE_FEW_AVAILABLE
+
+            return Variation.STATE_MANY_AVAILABLE
+
+        return None
+
+    def get_count_priority(self, event_time: EventTime):
+        scores = {}
+        info = {}
+
+        def count_severity(count, initial_amount, exp=0.5):
+            try:
+                return max(0, 1 - math.pow(count / initial_amount, exp))
+            except ArithmeticError:
+                return 1
+
+        if self.count == 0:
+            scores['depleted'] = 0
+        else:
+            now = event_time.datetime_to_event_time(datetime.now())
+            if self.count is not None and self.counted_at is not None:
+                # try to estimate the current count
+                total_sold = self.initial_amount - self.count
+                count_event_time = event_time.datetime_to_event_time(self.counted_at)
+                sale_rate = max(0, total_sold / max(1, count_event_time))
+                estimate = self.initial_amount - now * sale_rate
+                estimated_count = min(self.initial_amount, max(0, estimate))
+
+                scores['running_low_estimated'] = count_severity(estimated_count, self.count)
+                info['estimated_count'] = estimated_count
+                info['sale_rate'] = sale_rate * 3600
+
+                scores['running_low'] = 0.5 * count_severity(self.count, self.initial_amount)
+
+                count_age = max(0, now - count_event_time)
+                scores['outdated_count'] = 0.5 * math.pow(count_age / 3600 / 4.0, 0.5)
+                info['count_age'] = count_age
+            else:
+                scores['missing_count'] = 0.2
+
+
+            if self.count_prio_bumped:
+                scores['bumped'] = 1.0
```

Comment:
![image](https://github.com/user-attachments/assets/208ff802-4489-4711-9505-90526c5c187d)

My feeling is never counted items should be ranked higher than those with low count :thinking:

Reply:
You're right. There used to be a score that was added based on the count age, and never-counted items would rank higher over time. I think that still makes sense, as in the first few hours of the event, we don't have to count everything just so that it is counted. I will change it and treat never-counted items as if they were counted with their initial_amount at t=0. THen we always have those scores competing.

Uncounted items will now have this score behaviour over (event) time:

![image](https://github.com/user-attachments/assets/c50360e0-bd04-4fc9-bbe6-ee9bef6e89c7)